github/resource_github_repository_file.go

```diff
@@ -115,6 +115,24 @@ func resourceGithubRepositoryFile() *schema.Resource {
 				Description: "Enable overwriting existing files, defaults to \"false\"",
 				Default:     false,
 			},
+			"autocreate_branch": {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Description: "Automatically create the branch if it could not be found. Subsequent reads if the branch is deleted will occur from 'autocreate_branch_source_branch'",
+				Default:     false,
+			},
+			"autocreate_branch_source_branch": {
+				Type:        schema.TypeString,
+				Default:     "main",
+				Optional:    true,
```

Comment:
For both this attribute and the below one, Terraform offers a `RequiredWith` attribute that we can set to ensure that `autocreate_branch_source_sha` can't be used without `autocreate_branch`. Would you mind making that change?

Reply:
Added the `RequiredWith` option to the schema. Thanks @kfcampbell